File: FrontEnd-ProjectReact/my-react-app-inventory/src/types/rolePermission.ts
import type { Role } from "./role";

export interface RolePermission {
  id: number;
  roleId: number;
  permissionName: string;
  canView: boolean;
  canAdd: boolean;
  canEdit: boolean;
  canDelete: boolean;
  recStatus?: string | null;
  role: Role;
}

File: Sample/Sample.Infrastructure/EntityFramework/RolePermission.cs
using System;
using System.Collections.Generic;

namespace Sample.Infrastructure.EntityFramework
{
    public partial class RolePermission
    {
        public int Id { get; set; }
        public int RoleId { get; set; }
        public string PermissionName { get; set; } = null!;
        public bool CanView { get; set; }
        public bool CanAdd { get; set; }
        public bool CanEdit { get; set; }
        public bool CanDelete { get; set; }
        public string? RecStatus { get; set; }

        public virtual Role Role { get; set; } = null!;
    }
}
